src/sync.cpp

```diff
@@ -178,11 +168,7 @@ void AssertLockNotHeldInternal(const char* pszName, const char* pszFile, int nLi
 
 void DeleteLock(void* cs)
 {
-    LockData& lockdata = GetLockData();
-    if (!lockdata.available) {
-        // We're already shutting down.
-        return;
-    }
+    LockData& lockdata{*g_lockdata.Get()};
```

Comment:
(same above)
I prefer following code rather than calling Get() directly.
```suggestion
    LockData& lockdata = *g_lockdata;
```
or
```suggestion
    LockData& lockdata{*g_lockdata};
```